scripts/install_fast_mcljava_linux_mac.sh

```diff
@@ -38,28 +81,39 @@ java_inc=$1
   cd mcl
   git checkout $mcl_version || exit
   echo "----- Deleting currently installed version of mcl -----"
-  if [ $os == "linux" ]; then
+  if [ $os = "linux" ] || [ $os = "FreeBSD" ]; then
     sudo rm /usr/lib/libmcljava.so
   else # mac os
     mkdir -p ~/Library/Java/Extensions/ #check that this is included here: System.out.println(System.getProperty("java.library.path"));
 	  rm ~/Library/Java/Extensions/libmcljava.dylib
   fi
   echo "----- Building mcl -----"
-  make -j4 || exit # build mcl library
+  mkdir build 2>/dev/null
+  cd build
+  cmake .. -DCMAKE_C_FLAGS="${C_TUNING_FLAGS}" -DCMAKE_CXX_FLAGS="${C_TUNING_FLAGS}" -DMCL_STATIC_LIB="ON" 
+  cmake --build .
+  cd ..
+  export MCL_LIBDIR=$(pwd)/build/lib
   echo "----- Building mcl java bindings and running tests -----"
   echo "----- Java include path: $java_inc -----"
   cd ffi/java
-  make test_mcl JAVA_INC=-I$java_inc || exit # build java bindings, set include manually
+  mkdir build 2>/dev/null
+  cd build
+  export JAVA_HOME=$1/../
+  if [ -z ${2+x} ]; then
+	  cmake .. -DMCL_LINK_DIR=${MCL_LIBDIR} -DCMAKE_CXX_FLAGS="${C_TUNING_FLAGS} -I /usr/local/include"
```

Comment:
Had to add `-I /usr/local/opt/openssl/include` here and below for the compilation process to work on macOS.

Reply:
I think I forgot to specify a compiler flag in the FFI CMakeLists.
Can you check using ldd if the created libmcljava.so is linked against libcrypto or libssl? This is not specified by the CMakeLists or the original Makefile, but can happen transitively. 
If it does not link against these libraries, including the headers is unnecessary and might even be a potential problem. Since the declared functions cannot actually be linked, the linking the library might fail or there might even be run-time errors. If this is the case, I will fix the CMakeLists.txt, so it does not include these headers, making the change unnecessary.
Otherwise, I will add the compiler flag.

Reply:
```
 ~ % otool -L ~/Library/Java/Extensions/libmcljava.dylib
  ~/Library/Java/Extensions/libmcljava.dylib:
	@rpath/libmcljava.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/lib/libgmp.10.dylib (compatibility version 15.0.0, current version 15.1.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1200.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
```

Reply:
Can you try using  `-DMCL_DONT_USE_OPENSSL` instead of `-I /usr/local/opt/openssl/include` in line 104? If my assumption is correct, this should fix the compile issue.

Reply:
`cmake .. -DMCL_LINK_DIR=${MCL_LIBDIR} -DCMAKE_CXX_FLAGS="${C_TUNING_FLAGS} -I /usr/local/include -DMCL_DONT_USE_OPENSSL"`
yields 
```
[ 30%] Building CXX object CMakeFiles/mclelgamaljava.dir/elgamal_wrap.cxx.o
/Library/Developer/CommandLineTools/usr/bin/c++ -Dmclelgamaljava_EXPORTS -I/usr/local/Cellar/openjdk/17.0.1_1/libexec/openjdk.jdk/Contents/Home/include -I/usr/local/Cellar/openjdk/17.0.1_1/libexec/openjdk.jdk/Contents/Home/include/darwin -I/tmp/mcl/ffi/java/../../include -I/tmp/mcl/ffi/java/../../include/cybozu -I/tmp/mcl/ffi/java/../../external/cybozulib_ext/include -march=native -mtune=native -I /usr/local/include -DMCL_DONT_USE_OPENSSL -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -fPIC -MD -MT CMakeFiles/mclelgamaljava.dir/elgamal_wrap.cxx.o -MF CMakeFiles/mclelgamaljava.dir/elgamal_wrap.cxx.o.d -o CMakeFiles/mclelgamaljava.dir/elgamal_wrap.cxx.o -c /tmp/mcl/ffi/java/elgamal_wrap.cxx
In file included from /tmp/mcl/ffi/java/elgamal_wrap.cxx:243:
/tmp/mcl/ffi/java/../../include/cybozu/crypto.hpp:23:10: fatal error: 'openssl/hmac.h' file not found
#include <openssl/hmac.h>
         ^~~~~~~~~~~~~~~~
1 error generated.
```

It seems like `elgamal_wrap.cxx` (which we're not using, anyway, but it's built alongside the rest of the ffi stuff) includes `crypto.hpp`, which inevitably includes `openssl/hmac.h` [independently](https://github.com/herumi/mcl/blob/91ccb57b7b26538e18ec457afce1533165428443/include/cybozu/crypto.hpp#L23), as far as I can tell, of any flags.

Reply:
Now elgamal_wrap.cxx should not be built any more. I have changed the install script to only compile mcljava.

Reply:
That did the trick.